test/src/unit-cppapi-var-offsets.cc

```diff
@@ -605,4 +605,107 @@ TEST_CASE(
   VFS vfs(ctx);
   if (vfs.is_dir(array_name))
     vfs.remove_dir(array_name);
+}
+
+TEST_CASE(
+    "C++ API: Test 32-bit offsets",
+    "[var-offsets][sparse][32bit-offset][arrow]") {
+  std::string array_name = "test_32bit_offset";
+  create_sparse_array(array_name);
+
+  std::vector<int32_t> data = {1, 2, 3, 4, 5, 6};
+  // TODO: use a 32-bit offset vector when it's supported on the write path
```

Comment:
@Shelnutt2 FYI: @ypatia and I discussed this earlier this week: There isn't an obvious/useful way to make this 32-bit offsets work for the C++ API that uses `std::vector<uint64_t>`, so this is only supported for the `uin64_t*` API because recasting is easy.

This is already a hacky API that we've OK'd since we're going to mostly use it internally, so I think this is an acceptable limitation.

Reply:
@joe-maley that sounds fine to me, thanks for letting me know the limitation